Array_Problem_Task_Recursion.js

```diff
@@ -0,0 +1,48 @@
+const array2 = [
+  { id: "one", next: "two", value: null },
+  { id: "two", next: "three", value: null },
+  { id: "three", next: null, value: null },
+];
+const newArray = [];
+array2.forEach((element) => {
+  if (element.next === null) {
+    newArray.push(element);
+  } else {
+    let newobj = { ...element };
+    newArray.push(arrrayProblemFUnction(newobj));
+  }
+});
+function arrrayProblemFUnction(arrayobject) { //recursive Function
+  if (arrayobject.next !== null) {
+    let check = findingNextObject(arrayobject.next);
```

Comment:
##  findingNextObject( ) function is just checking and returning an object of given 'ID' if it exists in the array or does not exist. 
Code is complex because of this function only. As I did not use the arrays built-in function and implemented this function on my own.